src/planner/binder/expression/bind_star_expression.cpp

```diff
@@ -102,7 +112,8 @@ void Binder::ReplaceStarExpression(unique_ptr<ParsedExpression> &expr, unique_pt
 	    *expr, [&](unique_ptr<ParsedExpression> &child_expr) { ReplaceStarExpression(child_expr, replacement); });
 }
 
-static string ReplaceColumnsAlias(const string &alias, const string &column_name, optional_ptr<duckdb_re2::RE2> regex) {
+string Binder::ReplaceColumnsAlias(const string &alias, const string &column_name, void *regex_p) {
+	optional_ptr<duckdb_re2::RE2> regex((duckdb_re2::RE2 *)regex_p); // NOLINT
```

Comment:
Do we need to move the regex into these functions?

If we do, we can also forward declare to avoid needing to pass a void pointer.

Reply:
Great idea, completely forgot about forward declaring!

While on the subject, I'm not sure that `src/include/duckdb/function/scalar/regexp.hpp` should be including `re2/...`?
Only the cpp file should be doing that